lib/common/alarm/ScaleAlarms.ts

```diff
@@ -0,0 +1,143 @@
+import { Duration } from "aws-cdk-lib";
+import { ComparisonOperator, Stats } from "aws-cdk-lib/aws-cloudwatch";
+import { AlarmCloneFunction, AlarmWithAnnotation } from "../..";
+
+const SupportedMetricPeriods = [
+  Duration.minutes(1),
+  Duration.minutes(5),
+  Duration.minutes(15),
+  Duration.hours(1),
+  Duration.hours(6),
+  Duration.days(1),
+  Duration.days(7),
+];
+
+/**
+ * Properties for configuring a function that clones alarms to be scaled by multiplication factors.
+ */
+export interface ScaleAlarmsProps {
+  /**
+   * The disambiguator to assign to the alarms cloned by the function.
+   */
+  readonly disambiguator: string;
+
+  /**
+   * A multiplication factor to apply to the threshold of the cloned alarms.
+   * @default - 1.0
+   */
+  readonly thresholdMultiplier?: number;
+
+  /**
+   * A multiplication factor to apply to the datapointsToAlarm property of the cloned alarms.
+   * @default - 1.0
+   */
+  readonly datapointsToAlarmMultiplier?: number;
+
+  /**
+   * A multiplication factor to apply to the `evaluationPeriods` property of the cloned alarms.
+   * @default - Same as datapointsToAlarmMultiplier.
+   */
+  readonly evaluationPeriodsMultiplier?: number;
+}
+
+/**
+ * Creates a function that clones alarms to be more scaled using the multiplication factors
+ * provided. This function can be provided to the `cloneAlarms` method of
+ * {@link MonitoringFacade}.
+ *
+ * This can be used with scaling factors less than 1.0 to create more aggressive alarms, which
+ * can be useful to provide early detection and trigger change rollbacks.
+ *
+ * @param props The properties for configuring the function.
+ * @returns A function that clones alarms and applies scaling factors.
+ */
+export function ScaleAlarms(props: ScaleAlarmsProps): AlarmCloneFunction {
+  return function (alarm: AlarmWithAnnotation) {
+    let thresholdMultiplierToUse: number = 1;
+    if (props.thresholdMultiplier) {
+      const comparisonOperator =
+        alarm.alarmDefinition.addAlarmProps.comparisonOperator;
+      switch (comparisonOperator) {
+        case ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD:
+        case ComparisonOperator.GREATER_THAN_THRESHOLD:
+          thresholdMultiplierToUse = props.thresholdMultiplier;
+          break;
+        case ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD:
+        case ComparisonOperator.LESS_THAN_THRESHOLD:
+          thresholdMultiplierToUse = 1.0 / props.thresholdMultiplier;
+          break;
+        case ComparisonOperator.GREATER_THAN_UPPER_THRESHOLD:
+        case ComparisonOperator.LESS_THAN_LOWER_THRESHOLD:
+        case ComparisonOperator.LESS_THAN_LOWER_OR_GREATER_THAN_UPPER_THRESHOLD:
+          // Don't modify thresholds for anomoly detection models.
```

Comment:
The docs for `ScaleAlarms` should be more transparent about the expected behaviour with anomaly detection, otherwise it may be surprising.

Also:
```diff
- anomoly
+ anomaly
```